tests/testthat/test-model-compile.R

```diff
@@ -232,6 +233,55 @@ test_that("dir arg works for cmdstan_model and $compile()", {
   )
 })
 
+test_that("compiling stops on hyphens in stanc_options", {
+  skip_on_cran()
+  hyphens <- list("--allow-undefined")
+  hyphens2 <- list("--allow-undefined" = TRUE)
+  hyphens3 <- list("--o" = "something")
+  stan_file <- testing_stan_file("bernoulli")
+  expect_error(
+    cmdstan_model(stan_file, stanc_options = hyphens, compile = FALSE),
+    "No leading hyphens allowed in stanc options (--allow-undefined). Use options without leading hyphens, like for example `stanc_options = list('allow-undefined')`",
+    fixed = TRUE
+  )
+  expect_error(
+    cmdstan_model(stan_file, stanc_options = hyphens2, compile = FALSE),
+    "No leading hyphens allowed in stanc options (--allow-undefined). Use options without leading hyphens, like for example `stanc_options = list('allow-undefined')`",
+    fixed = TRUE
+  )
+  expect_error(
+    cmdstan_model(stan_file, stanc_options = hyphens3, compile = FALSE),
+    "No leading hyphens allowed in stanc options (--o). Use options without leading hyphens, like for example `stanc_options = list('allow-undefined')`",
+    fixed = TRUE
+  )
+  mod <- cmdstan_model(stan_file, compile = FALSE)
+  expect_error(
+    mod$compile(stanc_options = hyphens),
+    "No leading hyphens allowed in stanc options (--allow-undefined). Use options without leading hyphens, like for example `stanc_options = list('allow-undefined')`",
+    fixed = TRUE
+  )
+  expect_error(
+    mod$compile(stanc_options = hyphens2),
+    "No leading hyphens allowed in stanc options (--allow-undefined). Use options without leading hyphens, like for example `stanc_options = list('allow-undefined')`",
+    fixed = TRUE
+  )
+  expect_error(
+    mod$compile(stanc_options = hyphens3),
+    "No leading hyphens allowed in stanc options (--o). Use options without leading hyphens, like for example `stanc_options = list('allow-undefined')`",
+    fixed = TRUE
+  )
+})
+
+test_that("compiling works with only names in list", {
```

Comment:
This isn't new I just moved it up in the file to keep the compilation tests separate from check_syntax tests